core/src/avm2/regexp.rs

```diff
@@ -165,7 +214,12 @@ impl<'gc> RegExp<'gc> {
 
             re_match
         })?;
-
+        Some(re_match)
```

Comment:
nit: The result of `find_utf8_match_at()` can be returned directly without a `?` and the intermediate `re_match` variable.